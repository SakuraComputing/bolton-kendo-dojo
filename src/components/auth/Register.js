import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import TextFieldGroup from '../../common/TextFieldGroup';
import { registerUser } from '../../actions/authActions';

export class Register extends Component {

    state = {
        name: '',
        email: '',
        password: '',
        password2: '',
        errors: {}
    };

    onChange = (e) => {
        this.setState({[e.target.name]: e.target.value})
    };

    onSubmit = (e) => {
        e.preventDefault();
        const newUser = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password,
            password2: this.state.password2
        };
        this.props.registerUser(newUser, this.props.history);
    };

    render() {
        const { errors } = this.state;
        return (
            <div className="register">
                <div className="container">
                    <h1 className="form-title">Sign Up</h1>
                    <p className="form-text">Create your Members Account</p>
                    <form noValidate onSubmit={this.onSubmit}>
                        <TextFieldGroup
                            placeholder="Name"
                            name="name"
                            value={this.state.name}
                            type="name"
                            onChange={this.onChange}
                            error={errors.name}
                        />
                        <TextFieldGroup
                            placeholder="Email Address"
                            name="email"
                            value={this.state.email}
                            type="email"
                            onChange={this.onChange}
                            error={errors.email}
                        />
                        <TextFieldGroup
                            placeholder="Password"
                            name="password"
                            value={this.state.password}
                            type="password"
                            onChange={this.onChange}
                            error={errors.password}
                        />
                        <TextFieldGroup
                            placeholder="Confirm Password"
                            name="password2"
                            value={this.state.password2}
                            type="password"
                            onChange={this.onChange}
                            error={errors.password2}
                        />

                        <input type="submit" className="button button-info"/>
                    </form>
                </div>
            </div>
        );
    }
}

Register.propTypes = {
    registerUser: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
    auth: state.auth,
    errors: state.errors
});

export default connect(mapStateToProps, { registerUser })(withRouter(Register));
