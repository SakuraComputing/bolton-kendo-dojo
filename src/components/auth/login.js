import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginUser } from '../../actions/authActions';
import PropTypes from 'prop-types';
import TextFieldGroup from '../../common/TextFieldGroup';

export class Login extends Component {

  state = {
      email: '',
      password: '',
      errors: {}
  }

  onChange = (e) => {
      this.setState({ [e.target.name]: e.target.value })
  };

  onSubmit = (e) => {
      e.preventDefault();
      const userData = {
          email: this.state.email,
          password: this.state.password
      };
      this.props.loginUser(userData);
  }

  render() {

    const { errors } = this.state;

    return (
      <div>
        <div className="login">
            <div className="container">
                <div className="row">
                    <div className="col-md-8 m-auto">
                        <h1 className="login-title">Log In</h1>
                        <p className="login-text">Sign in to the Members Area</p>
                        <form onSubmit={this.onSubmit}>
                            <TextFieldGroup
                                placeholder="Email Address"
                                name="email"
                                value={this.state.email}
                                type="email"
                                onChange={this.onChange}
                                error={errors.email}
                            />
                            <TextFieldGroup
                                placeholder="Password"
                                name="password"
                                value={this.state.password}
                                type="password"
                                onChange={this.onChange}
                                error={errors.email}
                            />
                            <input
                                type="submit"
                                className="button button-info"/>
                        </form>
                    </div>
                </div>
            </div>
        </div>
      </div>
    )
  }
}

Login.propTypes = {
    loginUser: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
   auth: state.auth,
   errors: state.errors
});

export default connect(mapStateToProps, { loginUser })(Login);
