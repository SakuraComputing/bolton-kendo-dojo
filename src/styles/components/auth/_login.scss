.login {
    text-align: center;
}

.login-request {
    font-size: 110%;
    padding-top: 2em;
}

.email-address {
    width: 30rem;
    outline: none;
}

