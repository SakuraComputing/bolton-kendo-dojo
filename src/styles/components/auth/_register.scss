.register {
    height: 100vh;
    text-align: center;
}