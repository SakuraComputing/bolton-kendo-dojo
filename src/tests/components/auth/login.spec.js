import React from 'react';
import { Login } from '../../../components/auth/login';
import { shallow } from 'enzyme';

describe('Login Form', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Login />);
    })

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should set the state on change', () => {
        wrapper.find('TextFieldGroup').at(0).simulate('change', {
            target: { value: 'Change Function' }
        });
        expect(wrapper).toMatchSnapshot();
    });
    it('should call onSubmit the form', () => {
        // const p = Promise.resolve('success');
        const onSubmitSpy = jest.fn();

        const loginUser = jest.fn();

        wrapper = shallow(<Login loginUser={loginUser} onSubmit={onSubmitSpy}/>)
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        })

        expect(loginUser.mock.calls.length).toBe(1);
        
    });
});