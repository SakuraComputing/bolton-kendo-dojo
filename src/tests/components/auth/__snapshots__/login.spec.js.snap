// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-md-8 m-auto"
            >
              <h1
                className="display-4 text-center"
              >
                Log In
              </h1>
              <p
                className="lead text-center"
              >
                Sign in to your DevConnector account
              </p>
              <form>
                <TextFieldGroup
                  name="email"
                  placeholder="Email Address"
                  type="email"
                  value=""
                />
                <TextFieldGroup
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <input
                  className="btn btn-info btn-block mt-4"
                  type="submit"
                />
              </form>
            </div>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-md-8 m-auto"
            >
              <h1
                className="display-4 text-center"
              >
                Log In
              </h1>
              <p
                className="lead text-center"
              >
                Sign in to your DevConnector account
              </p>
              <form>
                <TextFieldGroup
                  name="email"
                  placeholder="Email Address"
                  type="email"
                  value=""
                />
                <TextFieldGroup
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <input
                  className="btn btn-info btn-block mt-4"
                  type="submit"
                />
              </form>
            </div>
          </div>
        </div>,
        "className": "login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-md-8 m-auto"
            >
              <h1
                className="display-4 text-center"
              >
                Log In
              </h1>
              <p
                className="lead text-center"
              >
                Sign in to your DevConnector account
              </p>
              <form>
                <TextFieldGroup
                  name="email"
                  placeholder="Email Address"
                  type="email"
                  value=""
                />
                <TextFieldGroup
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <input
                  className="btn btn-info btn-block mt-4"
                  type="submit"
                />
              </form>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-md-8 m-auto"
            >
              <h1
                className="display-4 text-center"
              >
                Log In
              </h1>
              <p
                className="lead text-center"
              >
                Sign in to your DevConnector account
              </p>
              <form>
                <TextFieldGroup
                  name="email"
                  placeholder="Email Address"
                  type="email"
                  value=""
                />
                <TextFieldGroup
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <input
                  className="btn btn-info btn-block mt-4"
                  type="submit"
                />
              </form>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="display-4 text-center"
                >
                  Log In
                </h1>,
                <p
                  className="lead text-center"
                >
                  Sign in to your DevConnector account
                </p>,
                <form>
                  <TextFieldGroup
                    name="email"
                    placeholder="Email Address"
                    type="email"
                    value=""
                  />
                  <TextFieldGroup
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <input
                    className="btn btn-info btn-block mt-4"
                    type="submit"
                  />
                </form>,
              ],
              "className": "col-md-8 m-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log In",
                  "className": "display-4 text-center",
                },
                "ref": null,
                "rendered": "Log In",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign in to your DevConnector account",
                  "className": "lead text-center",
                },
                "ref": null,
                "rendered": "Sign in to your DevConnector account",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <TextFieldGroup
                      name="email"
                      placeholder="Email Address"
                      type="email"
                      value=""
                    />,
                    <TextFieldGroup
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />,
                    <input
                      className="btn btn-info btn-block mt-4"
                      type="submit"
                    />,
                  ],
                  "onSubmit": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "email",
                      "onChange": undefined,
                      "placeholder": "Email Address",
                      "type": "email",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "password",
                      "onChange": undefined,
                      "placeholder": "Password",
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "btn btn-info btn-block mt-4",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-8 m-auto"
              >
                <h1
                  className="display-4 text-center"
                >
                  Log In
                </h1>
                <p
                  className="lead text-center"
                >
                  Sign in to your DevConnector account
                </p>
                <form>
                  <TextFieldGroup
                    name="email"
                    placeholder="Email Address"
                    type="email"
                    value=""
                  />
                  <TextFieldGroup
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <input
                    className="btn btn-info btn-block mt-4"
                    type="submit"
                  />
                </form>
              </div>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-8 m-auto"
              >
                <h1
                  className="display-4 text-center"
                >
                  Log In
                </h1>
                <p
                  className="lead text-center"
                >
                  Sign in to your DevConnector account
                </p>
                <form>
                  <TextFieldGroup
                    name="email"
                    placeholder="Email Address"
                    type="email"
                    value=""
                  />
                  <TextFieldGroup
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <input
                    className="btn btn-info btn-block mt-4"
                    type="submit"
                  />
                </form>
              </div>
            </div>
          </div>,
          "className": "login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col-md-8 m-auto"
              >
                <h1
                  className="display-4 text-center"
                >
                  Log In
                </h1>
                <p
                  className="lead text-center"
                >
                  Sign in to your DevConnector account
                </p>
                <form>
                  <TextFieldGroup
                    name="email"
                    placeholder="Email Address"
                    type="email"
                    value=""
                  />
                  <TextFieldGroup
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <input
                    className="btn btn-info btn-block mt-4"
                    type="submit"
                  />
                </form>
              </div>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-md-8 m-auto"
              >
                <h1
                  className="display-4 text-center"
                >
                  Log In
                </h1>
                <p
                  className="lead text-center"
                >
                  Sign in to your DevConnector account
                </p>
                <form>
                  <TextFieldGroup
                    name="email"
                    placeholder="Email Address"
                    type="email"
                    value=""
                  />
                  <TextFieldGroup
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <input
                    className="btn btn-info btn-block mt-4"
                    type="submit"
                  />
                </form>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="display-4 text-center"
                  >
                    Log In
                  </h1>,
                  <p
                    className="lead text-center"
                  >
                    Sign in to your DevConnector account
                  </p>,
                  <form>
                    <TextFieldGroup
                      name="email"
                      placeholder="Email Address"
                      type="email"
                      value=""
                    />
                    <TextFieldGroup
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                    <input
                      className="btn btn-info btn-block mt-4"
                      type="submit"
                    />
                  </form>,
                ],
                "className": "col-md-8 m-auto",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Log In",
                    "className": "display-4 text-center",
                  },
                  "ref": null,
                  "rendered": "Log In",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign in to your DevConnector account",
                    "className": "lead text-center",
                  },
                  "ref": null,
                  "rendered": "Sign in to your DevConnector account",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <TextFieldGroup
                        name="email"
                        placeholder="Email Address"
                        type="email"
                        value=""
                      />,
                      <TextFieldGroup
                        name="password"
                        placeholder="Password"
                        type="password"
                        value=""
                      />,
                      <input
                        className="btn btn-info btn-block mt-4"
                        type="submit"
                      />,
                    ],
                    "onSubmit": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "name": "email",
                        "onChange": undefined,
                        "placeholder": "Email Address",
                        "type": "email",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "name": "password",
                        "onChange": undefined,
                        "placeholder": "Password",
                        "type": "password",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "btn btn-info btn-block mt-4",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "form",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
